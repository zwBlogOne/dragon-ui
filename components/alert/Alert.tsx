import React, { Component } from 'react';
import Modal from '../modal';
import Button from '../button';
import Icon from '../icon';
import AlertProps from './PropsType';

class Alert extends Component<AlertProps, any> {
  static defaultProps = {
    prefixCls: 'ui-alert',
    theme: 'info',
    message: '',
    width: 270,
    className: '',
    hideIcon: false,
    closable: true,
    // closeText: '关闭',
    onClose: () => {},
  };

  render() {
    const {
      theme, message, closable, closeText, onClose, width,
      className, visible, prefixCls, hideIcon, locale,
    } = this.props;

    let iconType = 'info-round';
    switch (theme) {
      case 'warning':
        iconType = 'warning-round';
        break;
      case 'success':
        iconType = 'right-round';
        break;
      case 'error':
        iconType = 'wrong-round';
        break;
      default:
        break;
    }
    return (
      <Modal
        width={width}
        className={className}
        visible={visible}
      >
        <Modal.Body>
          <div className={prefixCls}>
            {!hideIcon && <Icon type={iconType} theme={theme} />}
            <span>{message}</span>
          </div>
        </Modal.Body>
        {
          closable && (
            <Modal.Footer>
              <Button onClick={onClose}>{closeText || locale!.close}</Button>
            </Modal.Footer>
          )
        }
      </Modal>
    );
  }
}

export default Alert;
