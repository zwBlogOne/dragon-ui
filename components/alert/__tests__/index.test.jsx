import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Alert from '../index';

describe('Alert', () => {
  it('renders basic Alert correctly', () => {
    const onClose = jest.fn();
    const wrapper = mount(
      <Alert
        theme="warning"
        onClose={onClose}
        message="这是一个警告框"
      />
    );
    wrapper.setProps({ visible: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
