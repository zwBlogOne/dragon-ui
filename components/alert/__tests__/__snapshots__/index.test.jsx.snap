// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders basic Alert correctly 1`] = `
<LocaleReceiver
  message="这是一个警告框"
  onClose={[Function]}
  theme="warning"
  visible={true}
>
  <AlertExtension
    className=""
    closable={true}
    hideIcon={false}
    locale={
      Object {
        "close": "关闭",
      }
    }
    localeCode="zh-cn"
    message="这是一个警告框"
    onClose={[Function]}
    prefixCls="ui-alert"
    theme="warning"
    visible={true}
    width={270}
  >
    <Modal
      animationDuration={300}
      animationType="zoom"
      className=""
      isRadius={false}
      isRound={false}
      minWidth={270}
      onMaskClick={[Function]}
      prefixCls="ui-modal"
      visible={true}
      width={270}
    >
      <div
        className="ui-modal fade-enter"
        onClick={[Function]}
        style={
          Object {
            "MozAnimationDuration": "300ms",
            "OAnimationDuration": "300ms",
            "WebkitAnimationDuration": "300ms",
            "animationDuration": "300ms",
            "msAnimationDuration": "300ms",
            "position": "fixed",
          }
        }
      >
        <div
          className="ui-modal-wrapper"
        >
          <div
            className="ui-modal-dialog zoom-enter"
            onClick={[Function]}
            style={
              Object {
                "MozAnimationDuration": "300ms",
                "OAnimationDuration": "300ms",
                "WebkitAnimationDuration": "300ms",
                "animationDuration": "300ms",
                "minWidth": 270,
                "msAnimationDuration": "300ms",
                "width": 270,
              }
            }
          >
            <ModalBody
              height="auto"
              prefixCls="ui-modal"
              style={Object {}}
            >
              <div
                className="ui-modal-body"
                style={
                  Object {
                    "height": "auto",
                  }
                }
              >
                <div
                  className="ui-alert"
                >
                  <Icon
                    className=""
                    onClick={[Function]}
                    prefixCls="ui-icon"
                    style={Object {}}
                    theme="warning"
                    type="warning-round"
                  >
                    <i
                      className="ui-icon ui-icon-warning-round theme-warning"
                      onClick={[Function]}
                      style={Object {}}
                    />
                  </Icon>
                  <span>
                    这是一个警告框
                  </span>
                </div>
              </div>
            </ModalBody>
            <ModalFooter
              prefixCls="ui-modal"
              style={Object {}}
            >
              <div
                className="ui-modal-footer"
                style={Object {}}
              >
                <Button
                  className={null}
                  isActive={false}
                  isBlock={false}
                  isCircle={false}
                  isDisabled={false}
                  isFocus={false}
                  isLoading={false}
                  isRadius={false}
                  isRound={false}
                  onClick={[Function]}
                  prefixCls="ui-button"
                  size={null}
                  style={Object {}}
                  theme="default"
                  type="button"
                >
                  <button
                    className="ui-button theme-default"
                    disabled={false}
                    onClick={[Function]}
                    style={Object {}}
                    type="button"
                  >
                    关闭
                  </button>
                </Button>
              </div>
            </ModalFooter>
          </div>
        </div>
      </div>
    </Modal>
  </AlertExtension>
</LocaleReceiver>
`;
